### ISSUE TITLE
Seed doesn't seem to be working for SKLearn interface

### ISSUE THREAD

**Issue by ledmaster** [Microsoft/LightGBM#116]
Hi,

I tried to run the sklearn_example.py file with different seeds, and I am getting the exact same results:

seed=0 -> The rmse of prediction is: 0.413481558002
Feature importances: [ 0.04766667  0.032       0.01633333  0.049       0.02333333  0.067       0.029
  0.01933333  0.00366667  0.04066667  0.03233333  0.02566667  0.00233333
  0.03433333  0.024       0.02066667  0.00333333  0.03633333  0.031
  0.02266667  0.00266667  0.05366667  0.07        0.02066667  0.07366667
  0.093       0.05366667  0.072     ]

seed=10 -> The rmse of prediction is: 0.413481558002
Feature importances: [ 0.04766667  0.032       0.01633333  0.049       0.02333333  0.067       0.029
  0.01933333  0.00366667  0.04066667  0.03233333  0.02566667  0.00233333
  0.03433333  0.024       0.02066667  0.00333333  0.03633333  0.031
  0.02266667  0.00266667  0.05366667  0.07        0.02066667  0.07366667
  0.093       0.05366667  0.072     ]

seed = 9876543 -> The rmse of prediction is: 0.413481558002
Feature importances: [ 0.04766667  0.032       0.01633333  0.049       0.02333333  0.067       0.029
  0.01933333  0.00366667  0.04066667  0.03233333  0.02566667  0.00233333
  0.03433333  0.024       0.02066667  0.00333333  0.03633333  0.031
  0.02266667  0.00266667  0.05366667  0.07        0.02066667  0.07366667
  0.093       0.05366667  0.072     ]

Thanks!

**Comment by wxchan** [comment]
@ledmaster seed is used in sample(when num of data>50000), bagging, sub_feature and tree dropout in dart, do you use any of these fields?

@guolinke did you rename seed in parameters?

**Comment by guolinke** [comment]
this actually is a bug, I forget to set seed in c_api. will fix soon.

**Comment by wxchan** [comment]
Before that, you can try to set other_params = {'bagging_fraction': 0.9, 'feature_fraction': 0.9} for example in fit() function.

**Comment by ledmaster** [comment]
@wxchan @guolinke 

Sure, thank you!

**Comment by guolinke** [comment]
@wxchan , I think we should use seed in sklearn to generate other-seed used in c_api.

**Comment by wxchan** [comment]
@ledmaster sorry, my mistake, it should be `other_params = {'bagging_seed': 1, 'feature_fraction_seed': 2}`

@guolinke I think simply set all other seeds equals to seed will be fine. If user wants to separate them, they can still use other_params.

**Comment by guolinke** [comment]
@wxchan , I use seed to generate other seeds inside c_api (only generate if seed is passed). And if user specific other seeds, the generate seeds will be overwritten.
